### PR TITLE
argon2 20151206 (new formula)

### PR DESCRIPTION
**Pull request by thibaultCha**
[Argon2](https://github.com/P-H-C/phc-winner-argon2) is the winner of the [Password Hashing Competition (PHC)](https://password-hashing.net/).

This Formula installs the library as well as the command-line utility.

### CODE REVIEW

File: Library/Formula/argon2.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  desc "The password hash Argon2, winner of PHC"
+  homepage "https://github.com/P-H-C/phc-winner-argon2"
+  url "https://github.com/P-H-C/phc-winner-argon2/archive/20151206.tar.gz"
+  sha256 "29f9eb6d84bfe29604c48830432aa030e96158d66e85a2c83b115e29eb025e85"
+
+  def install
+    system "make"
+    include.install "src/argon2.h"
+    lib.install "libargon2.dylib"
+    bin.install "argon2"
```
Comment (bfontaine) [inline Library/Formula/argon2.rb]: Is there no `make install` we could use?

File: Library/Formula/argon2.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Argon2 < Formula
+  desc "The password hash Argon2, winner of PHC"
```
Comment (bfontaine) [inline Library/Formula/argon2.rb]: We might want to shorten this to something like `"Password hash"`. We already know it’s called `argon2` and I’m not sure everybody knows what PHC is.

File: Library/Formula/argon2.rb (hunk 1 above)
Comment (thibaultCha) [inline Library/Formula/argon2.rb]: nope ([Makefile](https://github.com/P-H-C/phc-winner-argon2/blob/master/Makefile))

File: Library/Formula/argon2.rb (hunk 2 above)
Comment (thibaultCha) [inline Library/Formula/argon2.rb]: Sure. How does "Password hashing" sound?

Comment (thibaultCha) [inline Library/Formula/argon2.rb]: "Password hashing library and CLI utility" might be more descriptive.

Comment (bfontaine) [inline Library/Formula/argon2.rb]: :+1:

**Comment by thibaultCha** [comment]
@bfontaine Patch updated with the new description.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew, @thibaultCha; we appreciate it! :tada:
